packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestObjectProvider,
+    waitForContainerConnection,
+    ITestContainerConfig,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, ITestDataObject } from "@fluidframework/test-version-utils";
+
+/**
+ * Validates this scenario: When a datastore is aliased that it is considered a root datastore and always referenced
+ */
+describeNoCompat("GC DataStore Tombstoned", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let mainDataStore: ITestDataObject;
+
+    const sessionExpiryTimeoutMs = 1000;
+
+    const testContainerConfig: ITestContainerConfig = {
+        runtimeOptions: {
+            summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
+            gcOptions: {
+                gcAllowed: true,
+                sweepAllowed: true,
+                sessionExpiryTimeoutMs,
+                tombstoneMode: true,
+                snapshotCacheExpiryMs: 100,
+            },
+        },
+    };
+
+    async function waitForSummary(container: IContainer) {
+        const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+        return (dataStore._context.containerRuntime as ContainerRuntime).summarize(
+            { runGC: true, trackState: false, runSweep: true },
+        );
+    }
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        mainContainer = await provider.makeTestContainer(testContainerConfig);
+        mainDataStore = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    it("GC is notified when datastores are aliased.", async () => {
+        // Set a test value for the main root.
+        mainDataStore._root.set("testValue", "test");
+
+        await provider.ensureSynchronized();
+        // We run the summary so await this.getInitialSnapshotDetails() is called before the datastore is aliased
+        // and after the datastore is attached. This sets the isRootDataStore to false.
+        await waitForSummary(mainContainer);
+        mainDataStore._root.set("testValue2", "test2");
+        await provider.ensureSynchronized();
+        (mainDataStore._context as any).tombstone();
```

Comment:
Are you doing it this way because there is no way to configure sweep timeout today? If so, since Mark's change is in, you should update it.

This is more like a unit test and it should be in garbageCollection.spec.ts. Here, the end-to-end functionality should be test, i.e., it should wait for the sweep timeout to expire so that the data store is marked as tombstone and then validate that it cannot send ops, etc.

Reply:
Writing a test to do that now, I'll delete this test once I've updated it then